openqa-trigger-bisect-jobs

```diff
@@ -47,28 +48,60 @@ def call(cmds, dry_run=False):
 def openqa_clone(cmds, dry_run, default_opts=['--skip-chained-deps', '--within-instance'], default_cmds=['_GROUP=0']):
     return call(['openqa-clone-job'] + default_opts + cmds + default_cmds, dry_run)
 
+def fetch_url(url, request_type='text'):
+    if url.scheme == "file":
+        decodejson = lambda c: json.loads(c)
+        path = url.geturl()
+        path = path[7:]
+        path = 'test/data/python-requests/' + path
+        with open(path, "r") as request:
+            raw = content = request.read()
+    elif url.scheme in ["http", "https"]:
+        decodejson = lambda c: content.json()
+        try:
+            content = requests.get(url.geturl())
+        except requests.exceptions.RequestException as e:
+            log.error("Error while fetching "+url.geturl()+": "+str(e))
+            raise(e)
+        content.raise_for_status()
+        raw = r.content
+    if request_type == 'json':
+        try:
+            content = decodejson(content)
+        except json.decoder.JSONDecodeError as e:
+            log.error("Error while decoding JSON from "+url.geturl()+" -> >>"+raw+"<<: "+str(e))
+            raise(e)
+    return content
 
 def main():
     args = parse_args()
-    investigation_url = urljoin(args.url + '/', 'investigation_ajax')
-    log.debug('Retrieving investigation info from %s' % investigation_url)
-    out = requests.get(investigation_url)
-    log.debug('Received investigation info: %s' % out)
-    investigation = out.json()
-    os_test_issues = [line for line in investigation['diff_to_last_good'].split('\n') if 'OS_TEST_ISSUES' in line]
-    good, bad = [set(re.compile(': "([^"]*)",').search(line).group(1).split(',')) for line in os_test_issues]
+    parsed_url = urlparse(args.url)
+    investigation_url = parsed_url._replace(path=parsed_url.path + '/investigation_ajax')
```

Comment:
there is a reason why the method `_replace` is internal. We should not use it

Reply:
Do you have another idea how to set the path of `parsed_url`?
Like `$url->path('/foo/bar');` for Mojo::URL? I searched for it and didn't find anything.

Reply:
I've just read https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse and it only returns a named tuple but one cannot set the attributes manually. The docs even state the usage of `_replace` explicitly in one example so I suppose it isn't the worst to use it - you're just following an example from official documentation here.

Reply:
Maybe it is still cleaner to simply construct the `ParseResult` object you want directly, e.g.:

```
from urllib.parse import ParseResult
ParseResult(scheme='', netloc='', params='', query='', fragment='', path='/foo/bar').geturl()
```

Btw, that kind of construction is also mentioned in https://docs.python.org/3/librar/urllib.parse.html#urllib.parse.urlparse if you scroll down a bit.

Reply:
I want to mention https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlunparse regarding url construction.

Reply:
I replaced `_replace()` with one of the correct ways to do it (apparently there is more than one way!):
```
investigation_url = urlparse(urlunparse((parsed_url.scheme, parsed_url.netloc, parsed_url.path + '/investigation_ajax', '', '', '')))
```
beautiful!